### PR TITLE
Spark Tests

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/spark_test.go [hunk 1]
```diff
@@ -2880,3 +2883,428 @@ func createEMRAndS3(bucketName string) (SparkExecutor, SparkFileStore, error) {
...
+		BaseProvider BaseProvider
+	}
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   string
```
Comment (ahmadnazeri) [inline provider/spark_test.go:3204]: can we specify what these wants mean?

**Review by ahmadnazeri — APPROVED** [review]
LGTM; just one question.

File: provider/spark_test.go (hunk 1 above)
Comment (anthonylasso) [reply provider/spark_test.go:3204]: thought the same.

File: provider/spark_test.go [hunk 2]
```diff
@@ -2880,3 +2883,428 @@ func createEMRAndS3(bucketName string) (SparkExecutor, SparkFileStore, error) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			spark := &SparkOfflineStore{
+				Executor:     tt.fields.Executor,
+				Store:        tt.fields.Store,
+				Logger:       tt.fields.Logger,
+				query:        tt.fields.query,
+				BaseProvider: tt.fields.BaseProvider,
+			}
+			got, got1, got2 := spark.getResourceInformationFromFilePath(tt.args.path)
```
Comment (anthonylasso) [inline provider/spark_test.go:3298]: I'd match these identifiers to the return names. 
`fileType, fileName, fileVariant`

**Pull request merged**